src/main/java/com/conveyal/r5/streets/LinkedPointSet.java

```diff
@@ -411,44 +430,87 @@ public void makeStopToPointDistanceTables (Geometry treeRebuildZone) {
                 "Computed distances to PointSet points from {} of {} transit stops.");
         // Create a distance table from each transit stop to the points in this PointSet in parallel.
         // When applying a scenario, keep the existing distance table for those stops that could not be affected.
-        stopToPointDistanceTables = IntStream.range(0, nStops).parallel().mapToObj(stopIndex -> {
+        stopToPointLinkageCostTables = IntStream.range(0, nStops).parallel().mapToObj(stopIndex -> {
             Point stopPoint = transitLayer.getJTSPointForStopFixed(stopIndex);
             // If the stop is not linked to the street network, it should have no distance table.
             if (stopPoint == null) return null;
             if (treeRebuildZone != null && !treeRebuildZone.contains(stopPoint)) {
                 // This stop is not affected by the scenario. Return the existing distance table.
                 // All new stops created by a scenario should be inside the relink zone, so
                 // all stops outside the relink zone should already have a distance table entry.
-                if (stopIndex >= stopToPointDistanceTables.size()) {
+                if (stopIndex >= stopToPointLinkageCostTables.size()) {
                     throw new AssertionError("A stop created by a scenario is located outside relink zone.");
                 }
-                return stopToPointDistanceTables.get(stopIndex);
+                return stopToPointLinkageCostTables.get(stopIndex);
+            }
+
+            int[] linkageCostToPoints;
+
+            if (streetMode == StreetMode.WALK) {
+                // Walking distances from stops to street vertices are saved in the transitLayer.
+                // Get the pre-computed distance table from the stop to the street vertices,
+                // then extend that table out from the street vertices to the points in this PointSet.
+                TIntIntMap distanceTableToVertices = transitLayer.stopToVertexDistanceTables.get(stopIndex); // walk!
+                Envelope distanceTableZone = stopPoint.getEnvelopeInternal();
+                GeometryUtils.expandEnvelopeFixed(distanceTableZone, TransitLayer.WALK_DISTANCE_LIMIT_METERS);
+                linkageCostToPoints = distanceTableToVertices == null ? null :
+                        extendDistanceTableToPoints(distanceTableToVertices, distanceTableZone);
+
+            } else if (streetMode == StreetMode.BICYCLE) {
```

Comment:
A lot of code can be factored out of these mode-specific clauses. It wouldn't make the code shorter, but would serve to communicate that the same basic process is happening in both cases, just with some variables changed. For example, `StreetRouter sr = new StreetRouter(transitLayer.parentNetwork.streetLayer); sr.streetMode = streetMode; sr.quantityToMinimize = linkageCostUnit;` is the same in both cases.